sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceUtils.scala

```diff
@@ -93,6 +93,9 @@ object DataSourceUtils extends PredicateHelper {
    * in a driver side.
    */
   def verifySchema(format: FileFormat, schema: StructType, readOnly: Boolean = false): Unit = {
+    if (!SQLConf.get.isTimeTypeEnabled && schema.existsRecursively(_.isInstanceOf[TimeType])) {
+      throw QueryCompilationErrors.unsupportedTimeTypeError()
+    }
```

Comment:
Ditto. We don't have this for `Geo*Type`.

Reply:
no data source supports geo types yet, so it's not needed for now. But to be future-proof we should check geo here as well.